packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol

```diff
@@ -37,16 +37,34 @@ contract ArbTokenBridge is CloneFactory {
 
     ICloneable public immutable templateERC20;
     ICloneable public immutable templateERC777;
+    address immutable public l1Pair;
 
     modifier onlyEthPair {
         // This ensures that this method can only be called from the L1 pair of this contract
-        require(tx.origin == address(this), "ONLY_ETH_PAIR");
+        require(tx.origin == l1Pair, "ONLY_ETH_PAIR");
         _;
     }
 
-    constructor() public {
+    modifier onlyFromL2Token(address l1ERC20) {
+        // This ensures that this method can only be called by the L2 token
+        require(msg.sender == calculateBridgedERC777Address(l1ERC20)
+            || msg.sender == calculateBridgedERC20Address(l1ERC20)
+            || msg.sender == customToken[l1ERC20], "NOT_FROM_TOKEN");
+        _;
+    }
+    modifier onlyToL2Token(address l1ERC20, address to) {
+        // This ensures that this method can only be called by the L2 token
+        require(to == calculateBridgedERC777Address(l1ERC20)
+            || to == calculateBridgedERC20Address(l1ERC20)
+            || to == customToken[l1ERC20], "NOT_TO_TOKEN");
+        _;
+    }
+
+    constructor(address _l1Pair) public {
```

Comment:
why this change? (from `address(this)` to l1Pair as a param)?

Reply:
I created two subclasses, SymmetricBridge that initialises the constructor to `address(this)` and TestBridge that uses `msg.sender` so that we can mock messages coming from the core-bridge